### ISSUE TITLE
How could I reproduce the result for SQuAD 1.1?

### ISSUE THREAD

**Issue by alphaf52** [uwnlp/denspi#10]
Hi,

Thanks for your good work. I would like to reproduce the result for SQuAD 1.1 (as shown in Table 1 in the paper), but I am having some troubles. 

First, I downloaded the Pretrained Model from "gs://denspi/v1-0/model" and then tried to eval on dev-v1.1 using: "python run_piqa.py --do_predict --output_dir tmp --do_load --load_dir model --predict_file dev-v1.1.json --do_eval --gt_file dev-v1.1.json --metadata_dir bert"

The predicted answer seems to be random span, resulting in a metric like: {"exact_match": 0.47303689687795647, "f1": 4.43806570152543}. 0.47% EM means something is totally wrong.

I wonder whether I did it correctly.

And if I want to train a model to reproduce the result by myself, since I cannot get the Pretrained Model work, is it enough to just run the first step in the training section (i.e. "python run_piqa.py --train_batch_size 12 --do_train --freeze_word_emb --save_dir $SAVE1_DIR")

Thanks and hope to get your advice

**Comment by mittalpatel** [comment]
Hey @alphaf52 , could you find any solution for this? We are still facing the same issue.

**Comment by jhyuklee** [comment]
Hi, I think the problem is you forgot to give `--parallel`. The model is trained on DataParallel, so you have to give that option to load the model properly. Please try this and let me know.

**Comment by mittalpatel** [comment]
Thanks a lot @jhyuklee , this seems to be working!!! We provided --parallel while creating vectors and it is giving proper answers now. We are doing some further testing and will confirm of this soon.

Thanks once again for the hint. It really helped! 

@alphaf52 You may try this solution.

**Comment by yucoian** [comment]
> Hi, I think the problem is you forgot to give `--parallel`. The model is trained on DataParallel, so you have to give that option to load the model properly. Please try this and let me know.

@jhyuklee Hi, If I want to train a model to reproduce the result by myself,  how can I start to train a model from scratch to reproduce the result for SQuAD 1.1 (as shown in Table 1 in the paper)?  I also met the same trouble when I try to start all over again.
![image](https://user-images.githubusercontent.com/12205308/71154116-c469af80-2275-11ea-8394-2c6246b7a2ef.png)
How can I reproduce the result of DENSPI (dense only) with "EM=73,6 F1=81.7"  if I start from scratch？

**Comment by yucoian** [comment]
> 

I wonder whether you can reproduce the result of DENSPI (dense only) with "EM=73,6 F1=81.7" if you start from scratch？

**Comment by yucoian** [comment]
> Thanks a lot @jhyuklee , this seems to be working!!! We provided --parallel while creating vectors and it is giving proper answers now. We are doing some further testing and will confirm of this soon.
> 
> Thanks once again for the hint. It really helped!
> 
> @alphaf52 You may try this solution.

@mittalpatel , I wonder whether you can reproduce the result of DENSPI (dense only) with "EM=73,6 F1=81.7" if you start from scratch？

**Comment by yucoian** [comment]
> Hi, I think the problem is you forgot to give `--parallel`. The model is trained on DataParallel, so you have to give that option to load the model properly. Please try this and let me know.

@mittalpatel @jhyuklee  I wonder how can I train a model for scratch to reproduce the result  for SQuAD 1.1 (as shown in Table 1 in the paper) by myself? I want to reproduce the result from scratch.   Thanks a lot.

**Comment by yucoian** [comment]
our group can't reproduce the result for SQuAD 1.1 (as shown in Table 1 in the paper) from scratch either !
Please help ... @mittalpatel @jhyuklee @eunsol @mbforbes

**Comment by mittalpatel** [comment]
@yucoian at what point are you facing the problem? We could do it by following the steps given in the readme.

**Comment by yucoian** [comment]
Thank you very much! In  the "SQuAD v1.1 Experiments (Section 6.1)", I cannot  reimplement  the "DENSPI (dense only, with Coherency scalar)" model.  Could you please tell me how to adapt your released code to reproduce the  result of "DENSPI (dense only, with Coherency scalar)"? 
![image](https://user-images.githubusercontent.com/12205308/71607722-c2c5b300-2bb6-11ea-8e6a-87a87fa02fad.png)
